src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsService.java

```diff
@@ -156,6 +160,57 @@ public Optional<String> getWebHookUrl(String projectKey, String buildPlanId) {
         return Optional.of(jenkinsInternalUrlService.toInternalCiUrl(urlString));
     }
 
+    @Override
+    public void extractAndPersistBuildLogStatistics(ProgrammingSubmission programmingSubmission, ProgrammingLanguage programmingLanguage, ProjectType projectType,
+            List<BuildLogEntry> buildLogEntries) {
+        if (buildLogEntries.isEmpty()) {
+            // No logs received -> Do nothing
+            return;
+        }
+
+        if (programmingLanguage != ProgrammingLanguage.JAVA) {
+            // Not supported -> Do nothing
+            return;
+        }
+
+        ZonedDateTime jobStarted = getTimestampForLogEntry(buildLogEntries, ""); // First entry;
+        ZonedDateTime agentSetupCompleted = null;
+        ZonedDateTime testsStarted = null;
+        ZonedDateTime testsFinished = null;
+        ZonedDateTime scaStarted = null;
+        ZonedDateTime scaFinished = null;
+        ZonedDateTime jobFinished = buildLogEntries.get(buildLogEntries.size() - 1).getTime(); // Last entry
+        Integer dependenciesDownloadedCount = null;
+
+        // If the projectType is null, this is an old (Maven-only) exercise
+        if (projectType == null || projectType.isMaven()) {
+            agentSetupCompleted = getTimestampForLogEntry(buildLogEntries, "docker exec");
+            testsStarted = getTimestampForLogEntry(buildLogEntries, "Scanning for projects...");
+            testsFinished = getTimestampForLogEntry(buildLogEntries, "Total time:");
+            scaStarted = getTimestampForLogEntry(buildLogEntries, "Scanning for projects...", 1);
+            scaFinished = getTimestampForLogEntry(buildLogEntries, "Total time:", 1);
+            dependenciesDownloadedCount = countMatchingLogs(buildLogEntries, "Downloaded from");
+        }
+        else if (projectType.isGradle()) {
+            // agentSetupCompleted is not supported
+            testsStarted = getTimestampForLogEntry(buildLogEntries, "Starting a Gradle Daemon");
+            testsFinished = getTimestampForLogEntry(buildLogEntries,
+                    buildLogEntry -> buildLogEntry.getLog().contains("BUILD SUCCESSFUL in") || buildLogEntry.getLog().contains("BUILD FAILED in"));
+            scaStarted = getTimestampForLogEntry(buildLogEntries, "Task :checkstyleMain");
+            scaFinished = getTimestampForLogEntry(buildLogEntries,
+                    buildLogEntry -> buildLogEntry.getLog().contains("BUILD SUCCESSFUL in") || buildLogEntry.getLog().contains("BUILD FAILED in"), 1);
+            // dependenciesDownloadedCount is not supported
+        }
```

Comment:
I think we should log if it's neither Maven nor Gradle, since we then have to extend the functionality here.
Additionally you could prevent storing an invalid BuildLogStatistic with only null values in that case

Reply:
Good point, I added a log statement and returned without saving anything.